scripts/train/train.py

```diff
@@ -454,6 +458,16 @@ def main(cfg: DictConfig) -> Trainer:
         for name, logger_cfg in logger_configs.items()
     ] if logger_configs else None
 
+    # Adds mosaicml logger to composer if the run was sent from Mosaic platform, access token is set, and mosaic logger wasn't previously added
+    mosaicml_logger = None
+    if os.environ.get(MOSAICML_PLATFORM_ENV_VAR, 'false').lower(
+    ) == 'true' and os.environ.get(MOSAICML_ACCESS_TOKEN_ENV_VAR):
+        mosaicml_logger = MosaicMLLogger()
+        if loggers and not any(isinstance(x, MosaicMLLogger) for x in loggers):
+            loggers.append(mosaicml_logger)
+        else:
+            loggers = [mosaicml_logger]
```

Comment:
We will hit this statement if `loggers` exists and there is a preexisting MosaicMLLogger logger. This would be incorrect because we'd lose all the loggers and use a fresh mosaicmllogger

Reply:
i'd rewrite to something like:
```
loggers = [
        build_logger(str(name), logger_cfg)
        for name, logger_cfg in logger_configs.items()
    ] if logger_configs else []  # Make this a list instead of None so we don't have to do extra checks

if < mosaic env var checks, etc.>:
# Get the mosaicml_logger from loggers if possible
    mosaicml_logger = next(logger in loggers  if  isinstance(logger, MosaicMLLogger)), None)
    if mosaicml_logger is None:
        mosaicml_logger = MosaicMLLogger()
        loggers.append(mosaicml_logger)

```